### DISCUSSION TITLE
How to prevent garbage collection for helm releases?

### DISCUSSION THREAD

**Discussion by ableuler** [fluxcd/flux2#2373] [Q&A] [answered]
Thanks for this awesome project that we've been adopting in my team over the past months! 

However, now I'm running into a question which I don't find a solution for in the docs: Is it possible to prevent garbage collection for helm releases. Specifically, I want to be able to delete a `helmRelease` resource from the cluster without flux subsequently deleting the actual helm release. 

If I interpret the documentation [here](https://fluxcd.io/docs/components/kustomize/kustomization/#garbage-collection) correctly, the annotation `kustomize.toolkit.fluxcd.io/prune: disabled` on the `helmRelease` resource will prevent the `helmRelease` from being garbaged collected on deletion of a parent kustomization, which is useful but not what I am looking for here.

In the [helmRelease API reference](https://fluxcd.io/docs/components/helm/api/) I don't see any field that would allow me to disable pruning. Am I missing something or is this not yet supported?

**Comment by stefanprodan** [comment, marked as answer]
Before deleting a `HelmRelease` you can suspend it with `flux suspend hr <name>` then `flux delete hr <name>`, and the actual Helm release will not be garbage collected. You can do the same in Git by setting the `HelmRelease.spec.suspend` to `true`, then reconcile, then delete it from Git.

**Reply by artem-nefedov** [reply]
Is there a way to do it without completely disabling reconcile loop for helm, similar to how Kustomization's `prune: false` does it?

E.g. I know ahead of time that I will need to keep resources after deleting HelmRelease, but I still want updates for it to work properly.

**Comment by ableuler** [comment]
Thanks for the quick response. I think being able to set this as a property of `HelmRelease` similar to the case of a `Kustomization` might be useful, but it's definitely good to know of this trick.

**Comment by qudongfang** [comment]
does this trick work on `Kustomization` too?

**Reply by matheuscscp** [reply]
Yes: https://github.com/fluxcd/kustomize-controller/blob/60b8f86f490b46110d95079a49544de5580807c5/internal/controller/kustomization_controller.go#L988-L991
